pytest_fixtures/certificate.py

```diff
@@ -0,0 +1,150 @@
+from pathlib import Path
```

Comment:
This comment applies to the whole module.

Given how these fixture functions are defined (factory), and that the content of these functions is intended to be executed as test steps, not as setup, I think almost none of these should be fixtures.

They should largely be functions that are defined either on `broker.hosts.Host`, `robottelo.hosts.ContentHost`, `robottelo.hosts.Capsule`, or `robottelo.hosts.Satellite`.

Then you can limit certificate fixtures to functions that actually _call_ those instance methods on `broker.hosts.Host` based instances that the fixtures get from VmBroker contexts (or existing fixtures).  They would actually execute within the setup phase of the pytest runtest loop.

Your test functions would then call the relevant Host/ContentHost/Capsule/Satellite instance methods to execute the test steps of setting up certificates and installing Satellite.

This makes those functions available to the entire framework on an instance of the necessary class, to be used by any fixture or test function that needs to deal with certificates.